documentation/sphinx/source/authorization.rst

```diff
@@ -0,0 +1,119 @@
+#############
+Authorization
+#############
+
+.. warning :: Authorization is currently experimental and is not recommended for use in production.
+
+Introduction
+============
+
+:ref:`Multi-tenant <multi-tenancy>` database implies a couple of new concepts that did not previously exist in FoundationDB.
+The first is the concept of privilege levels: we have *data-plane clients* whose typical workload is limited to accessing a tenant keyspace.
+On the other hand, we have *control-plane clients* or *administrators* who may read or update cluster-wide configurations through system keyspace.
+These operations also include creation and deletion of tenants.
+The second is access control: with multiple tenant keyspaces, it comes naturally that we would want to restrict database access of a client to a subset of them.
+
+Privilege Levels
+----------------
+
+Authorization feature extends FoundationDB's existing TLS policy to distinguish administrators from data-plane clients,
+making TLS configuration a prerequisite for enabling authorization.
+There are only two privilege levels: *trusted* versus *untrusted* clients.
+Trusted clients are authorized to perform any operation that pre-authorization FoundationDB clients used to perform, including those accessing the system keyspace.
+Untrusted clients may only request what is necessary to access tenant keyspaces for which they are authorized.
+Untrusted clients are blocked from accessing anything in the system keyspace or issuing management operations that modifies the cluster in any way.
+
+In order to be considered a trusted client, a client needs to be :ref:`configured with a valid chain of X.509 certificates and a private key <configuring-tls>`,
+and its certificate chain must be trusted by the server.
+If the server was configured with trusted IP subnets, i.e. run with one or more ``--trusted-subnet-SUBNET_NAME`` followed by a CIDR block describing the subnet,
+then the client's IP as seen from the server must additionally belong to one of the subnets.
+
+Choosing to respond with an empty certificate chain during `client authentication <https://www.rfc-editor.org/rfc/rfc5246#section-7.4.6>`_,
+or not belonging to one of the trusted subnets marks the client as untrusted.
```

Comment:
I think this paragraph is a bit confusing. Just from reading this it might not be clear what a trusted subnet does. The paragraph above suggest that clients need to be in a trusted subnet AND provide a certificate, the next paragraph suggest that a client needs to be in a trusted subnet OR provide a valid certificate. Can you please clarify this a bit?

Reply:
Yes. I thought I would be `De Morgan's Law`-ing the whole thing but it comes off a bit ambiguous.
I'll rephrase it to address the untrusted cases more explicitly.